packages/headless-styles/sandbox/plugins/css-to-ts/convert-inputs/write-file.js

```diff
@@ -0,0 +1,58 @@
+import path from 'path'
+import { appendFileSync, existsSync, mkdirSync, writeFileSync } from 'fs'
+import indentObject from '../utils/indentLine'
+
+const generatedDisclaimer = `// This file is generated by running the Vite dev sever via
+// 'yarn start:sandbox' and **visiting localhost:3000**
+//
+// Manual changes will be lost - proceed with caution!
+\n
+`
+
+const writeToFile = (outPath, input, outputType) => {
+  let fileOutPath = outPath
+
+  // Case: "SplitFile": Export input object key, values as exports to separate file
+  if (outputType === 'splitFile') {
+    // Create dir for generated split files
+    if (path.extname(fileOutPath)) {
+      fileOutPath = fileOutPath.slice(0, -3)
+    } else {
+      fileOutPath = path.join(fileOutPath, input.filename)
+    }
+
+    if (!existsSync(fileOutPath)) {
+      mkdirSync(fileOutPath)
+    }
+
+    Object.entries(input.contents).forEach((key) => {
+      const body = indentObject(key[1], 2)
+      writeFileSync(path.join(fileOutPath, `${key[0]}.ts`), 'export default ')
+      appendFileSync(path.join(fileOutPath, `${key[0]}.ts`), body)
```

Comment:
I think it would be better to cache the path + filename into a variable here

```suggestion
      const fileOut = path.join(fileOutPath, `${key[0]}.ts`)
      writeFileSync(fileOut, 'export default ')
      appendFileSync(fileOut, body)
```